### PR TITLE
add task-4

### PR DESCRIPTION
**Pull request by spiderpoul**
1 - Task 4.1 is implemented
3 - TASK 4.2 is implemented lambda links are provided and returns data
4 - TASK 4.3 is implemented lambda links are provided and products is stored in DB (call TASK 4.2 to see the product)
5 - Frontend application is integrated with product service (/products API) and products from product-service are represented on Frontend. https://diw7ghb4t5jtk.cloudfront.net/

+1 - All lambdas return error 500 status code on any error (DB connection, any unhandled error in code)

FE link: https://diw7ghb4t5jtk.cloudfront.net/
endpoints:
GET - https://32wyze69d7.execute-api.eu-west-1.amazonaws.com/dev/products
GET - https://32wyze69d7.execute-api.eu-west-1.amazonaws.com/dev/products/1
POST - https://32wyze69d7.execute-api.eu-west-1.amazonaws.com/dev/products

### CODE REVIEW

File: product-service/services/index.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    throw new Error(`Product with id = [${id}] was not found`);
+  }
+  return product;
+}
+
+export const createProduct = async (productToCreate: Product) => {
+  const productQuery = {
+    text: `INSERT INTO
+        ${PRODUCTS_TABLE_NAME}(title, description, price)
+        VALUES($1, $2, $3) RETURNING *`,
```
Comment (KatyaAksenchik) [inline product-service/services/index.ts]: Maybe it is better to return just `id`, because only `id` we will need to create stock

**Comment by KatyaAksenchik** [comment]
Can you please add links to FE?

**Comment by Julia-ivv** [comment]
Also, please provide BE links

File: product-service/models/initDB.sql [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+create table if not exists products (
+	id uuid primary key default uuid_generate_v4(),
+	title text,
+	description text,
+	price integer,
+	imageUrl varchar(255)
+);
+
+create table if not exists stocks (
+	product_id uuid references products (id),
```
Comment (DzmitryU) [inline product-service/models/initDB.sql]: You could set `product_id` as a primary key for this table

**Comment by spiderpoul** [comment]
> Also, please provide BE links

Added
GET - https://32wyze69d7.execute-api.eu-west-1.amazonaws.com/dev/products
GET - https://32wyze69d7.execute-api.eu-west-1.amazonaws.com/dev/products/1
POST - https://32wyze69d7.execute-api.eu-west-1.amazonaws.com/dev/products

**Comment by spiderpoul** [comment]
> Can you please add links to FE?

addded
FE link: https://diw7ghb4t5jtk.cloudfront.net/

**Comment by KatyaAksenchik** [comment]
I wasn't able to create new product from FE. GET `products/${id}` return `{"message":"Internal Server Error"}`
For now scrore: 6/9

File: product-service/models/initDB.sql (hunk 2 above)
Comment (spiderpoul) [reply product-service/models/initDB.sql]: thanks for your review, fixed

File: product-service/services/index.ts (hunk 1 above)
Comment (spiderpoul) [reply product-service/services/index.ts]: thanks for your review, fixed

**Pull request merged**